enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,172 @@
+The CLI in-cluster manager will gather information from each CLI's CR about each CLI and its image.  A Krew index will be populated in every cluster to provide available CLIs and plugins for a cluster.
+The logic for `oc image extract` can be re-used and extended for extracting the CLI binaries.  CLIs will be installed and managed by users through Krew, modified and extended to handle OpenShift CLIs.    
+
+Example function to be provided by Option 4 above (that will be accomplished through CLI CustomResources, CLI image, and Krew):
+
+```console
+$ oc image extract quay.io/sallyom/kn:latest --path /tmp/amd64/kn.tar.gz:$HOME/openshift-clis/compressed/
+$ tar xvf $HOME/openshift-clis/compressed/kn.tar.gz -C $HOME/openshift-clis/
+```
+
+As laid out in this proposal, each CLI will provide a separate image.
+Each CLI is responsible for creating a CR to hold metadata that `oc krew` will depend on.  The CR will serve to deliver the metadata and description
+of its deliverable binary.  An in-cluster CLI manager controller will access the CRs to populate and maintain an in-cluster Krew Index.  Users will install OpenShift tools that are known compatible
+with each cluster version through `oc krew`.    
+
+A CLI must provide a ClusterCLI CR.  The result of this proposal will be:
+* ClusterCLI Custom Resource Definition
+* Krew Index and a controller to create the index from CLI CRs.
+* mechanism to manage CLIs via ClusterCLI CRs and Krew
+* a reference consumer implementation, oc, that other CLIs can follow
+
+The CLI in-cluster management will require:
+- CLI images in an available registry- each CLI will produce an image to deliver its artifacts
+- Consumers - through an in-cluster Krew Index and the krew command, consumers will manage OpenShift CLIs
+  Consumers will require access to CLI images.  In disconnected environments, this will be a mirrored local registry. 
+- Producers - each CLI that wants to offer its binaries in-cluster is responsible for providing its metadata and description via a ClusterCLI as well as
+  a CLI image to deliver its artifacts. 
+
+### Non-Goals
+
+* CLI manager will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism. 
+* CLI manager will not create or update the ClusterCLI Custom Resources, those will be managed by individual CLIs and their operators. 
+
+The ClusterCLI CRD will be installed either by a new lightweight operator or by an existing operator.
+The ClusterCLI/oc Custom Resource will serve as a reference consumer implementation for other CLIs to follow.    
+
+[for an example I've created a ClusterCLI controller that will populate 3 test CRs, oc, odo, and kn](https://github.com/sallyom/clustercli)
+
+
+## Proposal
+
+### User Stories
+
+#### Story 1
```

Comment:
I want a clear list of commands users will use:
- user invokes `oc krew list` to learn installed plugins
- user invokes `oc krew search` which communicates with the krew index served by the cli-manager operator and gets information about CLIs
- user invokes `oc krew install foo` and gets `oc foo`

Reply:
added, updated